cumulusci/utils/yaml/cumulusci_yml.py

```diff
@@ -1,4 +1,276 @@
+    slug: str = None
+    is_listed: bool = None
+    steps: Dict[str, Union[FlowReference, TaskReference]]
+    checks: List[Check] = None
+    group: str = None
+    error_message: str = None
+    error_message: str = None
+    post_install_message: str = None
+    preflight_message: str = None
+
+
+class Project(CCIDictModel):
+    name: Optional[str]
+    package: Package = None
+    test: Test = None
+    git: Git = None
+    dependencies: List[Dict[str, URL]] = None  # TODO
+    apexdoc: ApexDoc = None
+    source_format: Literal["sfdx", "mdapi"] = "mdapi"
+
+
+class ScratchOrg(CCIDictModel):
+    config_file: Path
+    days: int = None
+    namespaced: str = None
+
+
+class Orgs(CCIDictModel):
+    scratch: Dict[str, ScratchOrg]
+
+
+class Attribute(CCIDictModel):
+    description: str
+    required: bool
+
+
+class Service(CCIDictModel):
+    description: str
+    attributes: Dict[str, Attribute]
+    validator: PythonClassPath = None
+
+
+class CumulusCIConfig(CCIDictModel):
+    keychain: PythonClassPath
+
+
+class Source(CCIDictModel):
+    github: URL = None
+    release: str = None
+
+
+class CumulusCLIConfig(CCIDictModel):
+    show_stacktraces: bool = None
+    plain_output: bool = None
+
+
+class CumulusCIRoot(CCIDictModel):
+    tasks: Dict[str, Task] = {}
+    flows: Dict[str, Flow] = {}
+    project: Project = {}
+    orgs: Orgs = {}
+    services: Dict[str, Service] = {}
+    cumulusci: CumulusCIConfig = None
+    plans: Dict[str, Plan] = []
+    minimum_cumulusci_version: str = None
+    sources: Dict[str, Source] = []
+    cli: CumulusCLIConfig = None
+
+
+class CumulusCIFile(CCIDictModel):
+    __root__: CumulusCIRoot
+
+
+def parse_from_yaml(source):
+    "Parse from a path, url, path-like or file-like"
+    return CumulusCIFile.parse_from_yaml(source)
+
+
+def validate_data(
+    data: Union[dict, list],
+    context: str = None,
+    on_error: callable = None,
+):
+    """Validate data which has already been loaded into a dictionary or list.
+
+    context is a string that will be used to give context to error messages.
+    on_error will be called for any validation errors with a dictionary in Pydantic error format
+
+    https://pydantic-docs.helpmanual.io/usage/models/#error-handling
+    """
+    return CumulusCIFile.validate_data(data, context=context, on_error=on_error)
+
+
+class ErrorDict(TypedDict):
+    "The structure of a Pydantic error dictionary. Google TypedDict if its new to you."
+    loc: Sequence[Union[str, int]]
+    msg: str
+    type: str
+
+
+def _log_yaml_error(logger, error: ErrorDict):
+    "Format and log a Pydantic-style error dictionary"
+    logger.warning("CumulusCI Parsing Error:")
+    loc = " -> ".join(repr(x) for x in error["loc"] if x != "__root__")
+    logger.warning("%s : %s", loc, error["msg"])
+    logger.error(
+        "NOTE: These errors will cause major problems in future versions of CumulusCI."
+    )
+    logger.error("Please correct them before May 1, 2021.")
```

Comment:
Let's wait and see how many errors there are in the wild before committing to a timeline.